tests/examples.lean

```diff
@@ -109,3 +110,46 @@ begin
 end
 
 end refine_struct
+
+/- traversable -/
+
+meta def check_defn (n : name) (e : pexpr) : tactic unit :=
+do (declaration.defn _ _ _ d _ _) ← get_decl n,
+   e' ← to_expr e,
+   guard (d =ₐ e') <|> trace d >> failed
+
+set_option trace.app_builder true
+
+@[derive traversable]
+structure my_struct (α : Type) :=
+  (y : ℤ)
+
+run_cmd do
+check_defn ``my_struct.traversable
+  ``( { traversable .
+        to_functor := my_struct.functor,
+        traverse := λ (m : Type → Type) (_inst : applicative m) (α β : Type) (f : α → m β) (x : my_struct α),
+               my_struct.rec (λ (x : ℤ), pure (λ (a : ulift ℤ), {y := a.down}) <*> pure {down := x}) x} )
+
+@[derive traversable]
+structure my_struct2 (α : Type u) : Type u :=
+  (x : α)
+  (y : ℤ)
+  (z : list α)
+  (k : list (list α))
+
+run_cmd do
+check_defn ``my_struct2.traversable
+  ``( { traversable .
+        to_functor := my_struct2.functor,
+        traverse := λ (m : Type u → Type u) (_inst : applicative m) (α β : Type u) (f : α → m β) (x : my_struct2 α),
+               my_struct2.rec
+                 (λ (x_x : α) (x_y : ℤ) (x_z : list α) (x_k : list (list α)),
+                    pure
+                              (λ (a : β) (a_1 : ulift ℤ) (a_2 : list β) (a_3 : list (list β)),
+                                 {x := a, y := a_1.down, z := a_2, k := a_3}) <*>
+                            f x_x <*>
+                          pure {down := x_y} <*>
+                        traverse f x_z <*>
+                      traverse (traverse f) x_k)
+                 x } )
```

Comment:
Do we need this? It looks a bit messy of a check. Surely it suffices just to check that it has indeed been defined and has the right type; there aren't many possible ways the definition could work otherwise

Reply:
It is a bit messy but it tests several features of the derivation. I think it is worth having at least to signal us when those details change.

Reply:
Okay. In that case, you should clean up the formatting a bit so it looks more like a handwritten term, give the variables proper names and so on.